### ISSUE TITLE
subtitles not working (and quick fix)

### ISSUE THREAD

**Issue by user440** [popcorntime/popcorntime#603]
Operating System Version:
win 7

Popcorn Time Version:
0.3.10 build 252

Download date:
today

#### Expected Behaviour
to see subtitles

#### Actual Behaviour
no subtitles on every title. removed and installed. tried to change user-agent into both opensubtitles.js without any change.

**Comment by RoToRx88** [comment]
It seems that during a period of time everyday the subs are not working, or randomly.
I also have this problem between around 7pm and 10pm UTC+2 usually.
Maybe it is related to an overload from opensubtitles ?
Sometimes, I just stop watching, and restart, then the subs are downloaded.

**Comment by user440** [comment]
It seems that I have found whats the problem. It is a combination of number of open DHT connections and popcorn not hooking subtitles when they are loaded during the play.

When there are over 100 DHT open connections this causes a problem (on some systems?) that most of additional requests get timed out including DHT and subtitle requests. Upon checking console output there was on numerous occasions:

[ERROR] subtitleProvider.fetch() Error: connect ETIMEDOUT 104.25.132.104:443(…)
   win.error | @ | app.js:21
   (anonymous function) | @ | streamer.js:450

Subtitle missing issue existed in previous popcorn versions but I have fixed it back then indirectly by setting DHTlimit to 5 but now in the latest popcorn build DHTLimit setting is ignored so missing subtitle issue appeared again but I have edited popcorn code and it is working again.

So for a quick fix edit "src/app/lib/streamer.js" and "src/app/vendor/videojshooks.js" files as in my code change proposal at
https://github.com/popcorn-official/popcorn-desktop/pull/609/files

Save changes and restart popcorn. With this fix popcorn will retry up to 4 times to get subtitles.
